src/pages/issues.tsx

```diff
@@ -9,36 +8,88 @@ import {
     NO_ISSUES_FOUND_MESSAGE,
 } from '@/components/constants/messages';
 import { ISSUES_URL } from '@/components/constants/url';
+import axios from 'axios';
+
+type SearchFieldProps = {
+    searchText: string;
+    onSearchTextChanged: (event: any) => void;
+    onSearchTextSubmitted: () => void;
+    loading: boolean;
+};
+
+const SearchField = ({
+    searchText,
+    onSearchTextChanged,
+    onSearchTextSubmitted,
+    loading,
+}: SearchFieldProps) => {
+    return (
+        <div className={classNames.searchFieldContainer}>
+            <input
+                placeholder="Enter query string to search issues"
+                value={searchText}
+                onChange={onSearchTextChanged}
+                className={classNames.issueSearchInput}
+            />
+            <button
+                onClick={onSearchTextSubmitted}
+                className={classNames.issuesSearchSubmitButton}
+                disabled={loading || !searchText.trim()}
+            >
+                Submit
+            </button>
+        </div>
+    );
+};
 
 const Issues: FC = () => {
     const [issueList, setIssueList] = useState<[]>([]);
+    const [searchText, setSearchText] = useState('');
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<null | any>(null);
 
-    const { response, error, isLoading } = useFetch(ISSUES_URL, {
-        data: {
-            userData: {
-                roles: {
-                    super_user: true,
-                },
-            },
-        },
-    });
+    const fetchIssues = async () => {
+        try {
+            setIsLoading(true);
+            const res = await fetch(`${ISSUES_URL}?q=${searchText}`);
+            const data = await res.json();
+            if ('issues' in data) {
+                const issuesAndPullRequests: any = data.issues;
+                const onlyIssues = issuesAndPullRequests.filter(
+                    (issue: { hasOwnProperty: (arg0: string) => any }) =>
+                        !Object.prototype.hasOwnProperty.call(
```

Comment:
wait do we really need to do this & why ?

Reply:
Yes. GitHub issues API treats both PRs and issues as issues.

Kindly refer: https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#list-organization-issues-assigned-to-the-authenticated-user

Reply:
I meant to say do we really need to use `Object.hasownProperty` with the `call` method?

Reply:
Yes, as es-lint throws the following error: `Do not access Object.prototype method 'hasOwnProperty' from target object.eslint[no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins)`
Kindly refer: https://eslint.org/docs/latest/rules/no-prototype-builtins#rule-details

Reply:
Thanks for this read